pkg/controllers/infracluster/infracluster_controller_test.go

```diff
@@ -0,0 +1,266 @@
+package infracluster
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-capi-operator/pkg/operatorstatus"
+	"github.com/openshift/cluster-capi-operator/pkg/test"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	awsv1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ = Describe("InfraCluster", func() {
+	ocpInfraAWS := configv1.Infrastructure{
+		Status: configv1.InfrastructureStatus{
+			APIServerInternalURL: "https://api-int.test-cluster.test-domain:6443",
+			Platform:             configv1.AWSPlatformType,
+			PlatformStatus: &configv1.PlatformStatus{
+				Type: configv1.AWSPlatformType,
+				AWS: &configv1.AWSPlatformStatus{
+					Region: "us-east-1",
+				},
+			},
+		},
+	}
+
+	infraClusterWithExternallyManagedByAnnotation := &awsv1.AWSCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultInfraClusterName,
+			Namespace: defaultCAPINamespace,
+			Annotations: map[string]string{
+				clusterv1.ManagedByAnnotation: managedByAnnotationValueClusterCAPIOperator,
+			},
+		},
+	}
+
+	// When the annotation is present and the value is empty, it is CAPI owned. When the annotation is
+	// present with any other value, it is not CAPI owned.
+	thirdPartyAnnotation := "thirdparty"
+	infraClusterWithExternallyManagedByAnnotationWithValue := &awsv1.AWSCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultInfraClusterName,
+			Namespace: defaultCAPINamespace,
+			Annotations: map[string]string{
+				clusterv1.ManagedByAnnotation: thirdPartyAnnotation,
+			},
+		},
+	}
+
+	infraClusterWithoutExternallyManagedByAnnotation := &awsv1.AWSCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultInfraClusterName,
+			Namespace: defaultCAPINamespace,
+		},
+	}
+
+	Context("When there is no InfraCluster", func() {
+		var ic *awsv1.AWSCluster
+		BeforeEach(func() {
+			ic = infraClusterWithExternallyManagedByAnnotation.DeepCopy()
+			mustReconcile(ocpInfraAWS, ic, true)
+		})
+		AfterEach(func() {
+			Expect(test.CleanupAndWait(ctx, cl, ic)).To(Succeed())
```

Comment:
The argument for the `ic` here should just be an empty version of the object, you don't need to pass specific objects. It is designed to clean up all instances of the type

Reply:
It is not how `test.CleanupAndWait` works as this is a local package.
What you are describing is `cluster-api-actuator-pkg`'s `testutils.CleanupResources` behaviour, which also deletes the namespace after deleting the resource belonging to it. We refactored to use that instead.